contracts/tests/MockStakingRewards.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.4;
+
+import {SafeMath} from "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {
+    SafeERC20
+} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract MockStakingRewards {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    IERC20 public stakingToken;
+
+    uint256 private _totalSupply;
+    mapping(address => uint256) private _balances;
+
+    constructor(address _stakingToken) {
+        require(
+            _stakingToken != address(0),
+            "Staking token must be non-zero address"
+        );
+        stakingToken = IERC20(_stakingToken);
+    }
+
+    function totalSupply() external view returns (uint256) {
+        return _totalSupply;
```

Comment:
I don't really see a reason to make `_totalSupply` private and then have a function to return it.

Reply:
Fixed here: https://github.com/ribbon-finance/ribbon-v2/pull/194/commits/92bfe9b03ee5c687cb20a46e179a713040d6aee6